### DISCUSSION TITLE
Migrating generic test fixtures from NUnit

### DISCUSSION THREAD

**Discussion by MineCake147E** [thomhurst/TUnit#1277] [Q&A]
NUnit has a feature that allows us to define generic test fixtures.
https://docs.nunit.org/articles/nunit/writing-tests/attributes/testfixture.html#generic-test-fixtures
It can be used for testing multiple implementations of the same set of interfaces in a single class, without duplicating any code for different implementations.
Is there any way to migrate such tests?

**Comment by thomhurst** [comment]
Passing in different types of data is meant to do the trick, but I haven't tested generic classes that thoroughly to be honest.

Try using method data sources or argument attributes with different types and let me know how you get on?

Might need to make some changes
